packages/mention/src/withMention.ts

```diff
@@ -115,10 +115,11 @@ export const withMention = <
       )
     );
 
+    const previousCharPattern = triggerPreviousCharPattern ?? /\s/;
```

Comment:
We can move the default pattern into to `createMentionPlugin` and change it to `/\s?/`. This lets us remove the `beginningOfLine` check since this is incorporated into the regex.

Reply:
sure!

Reply:
I used `/^$|^\s$/` because `/\s?/` matched literally anything.

`/^$|^\s$/` is: "match an empty string, or match a space". Also modified documentation to explain what is the default behaviour.

Reply:
Good catch. I think `/^\s?$/` would be equivalent and a little easier to read.

Can you add the default pattern in the `options` object [here](https://github.com/udecode/plate/blob/bf8d5c82cfe24081cc279021edaf4bb833bdf263/packages/mention/src/createMentionPlugin.ts#L25) and remove it from [withMention.ts](https://github.com/udecode/plate/pull/2542/files#diff-8bfaad03ff7ba982d1c86122af02ca4d8c1ed9bea157783e598aa5c465dc4902R118)? Plate will automatically use it if no other value is specified, so no need for the `??` operator.

Reply:
Ah, definitely easier to read.

Sure, will do. But 

`const matchesPreviousCharPattern = triggerPreviousCharPattern.test(previousChar);`

triggers `'triggerPreviousCharPattern' is possibly 'undefined'.`

should I just add the safe accessing? I mean:

`const matchesPreviousCharPattern = triggerPreviousCharPattern?.test(previousChar);`

Reply:
The `?.` is fine, yes. We do a similar thing [here](https://github.com/udecode/plate/blob/d359bc959697d614c0d338577e6743595d3fb61f/packages/mention/src/getMentionOnSelectItem.ts#L62C38-L62C38). Thanks for making those changes! ❤️

Reply:
Thank you for your kind review :)